v1/src/test/java/com/google/cloud/teleport/templates/PubsubToBigQueryLT.java

```diff
@@ -124,49 +148,42 @@ public void testBacklog10gb() throws IOException, ParseException, InterruptedExc
     Result result =
         pipelineOperator.waitForConditionAndFinish(
             createConfig(info, Duration.ofMinutes(40)),
-            () -> waitForNumMessages(info.jobId(), INPUT_PCOLLECTION, numMessages));
+            () ->
+                bigQueryResourceManager.getRowCount(PROJECT, table.getDataset(), table.getTable())
+                    == NUM_MESSAGES);
```

Comment:
`>= NUM_MESSAGES` is safer here due to Beam/Dataflow not offering exactly-once guarantees in some cases.

Reply:
Done